theano/scan_module/scan_op.py

```diff
@@ -402,6 +402,32 @@ def make_node(self, *inputs):
                     'by using dimshuffle or shape_padleft. '
                     )
 
+        def _check_broadcast(v1, v2):
+            """ Checks that the broadcast pattern of v1 and v2.
+
+            Controls that the broadcast pattern of the variable provided as
+            input to `scan` matches the broadcast pattern provided in
+            `output_info`. It raises an error when they don't match. The
+            typical case is when the user provides either the input or the
+            `output_info` (but not both) with a dimension fixed to 1,
+            which may wrongly be interpreted as broadcastable.
+
+            """
+            msg = ("The broadcast pattern of the output of scan (%s) is "
+                   "inconsistent with the one provided in `output_info` "
+                   "(%s). The output on axis %d is `%r`, but it is `%r` on "
+                   "axis %d in `output_info`. This can happen if one of the "
+                   "dimension is fixed to 1 in the input, while it is still "
+                   "variable in the output, or vice-verca. You have to make "
+                   "them consistent.")
+            size = min(len(v1.broadcastable), len(v2.broadcastable))
```

Comment:
Some of the errors are due to the fact that this check was done on other stuff then the output_info, including non_sequence that can be anything. This mean v1 and v2 do not always have broadcastable. I don't remember if moving the check to only output_info will fix the problem or not. In all cases, skiping the check if the broadcastable attribute isn't there would fix this.